plotly/plotly/plotly.py

```diff
@@ -474,7 +474,7 @@ def get_figure(file_owner_or_url, file_id=None, raw=False):
 
     if raw:
         return figure
-    return tools.get_valid_graph_obj(figure, obj_type='Figure')
```

Comment:
@chriddyp @jmmease 
The purpose of this commit is to refactor `get_valid_graph_obj` function to _not_ remove invalid properties of a figure when `.get_figure()` is called. I think this is important because:
-it is consistent with Jon's preference to validation over coercing to default attributes as a default (which I agree with)
-if a user does want to bypass validation errors, setting `raw=True` is still an option in the get_figure function 
-one step on the road to finishing https://github.com/plotly/plotly.py/issues/283 fwiw

What do y'all think?

Reply:
I like this. The user will either get a `dict` with whatever was on the server (`raw=True`) or a fully valid `Figure`.

Do you think the stripping of invalid properties is something that regular users are going to need? I'm not clear on why we end up with figures with invalid properties on the server in the first place.

Reply:
I wondered the same thing. In refactoring these tests I noticed that a few of the example figures I pulled from were 4 years old and contained params that are no longer part of the valid params.

Reply:
I'd be inclined to update the test data, but I'll leave it to you and @chriddyp to decide whether we need to support some kind of legacy property removal.